addons/website_sale/views/templates.xml

```diff
@@ -156,6 +153,10 @@
                 </div>
                 <div class="o_wsale_product_btn"/>
             </div>
+            <t t-set="color" t-value="td_product['ribbon']['color'] or ''"/>
+            <t t-set="text_color" t-value="td_product['ribbon']['text_color'] or ''"/>
```

Comment:
```suggestion
            <t t-set="text_color" t-value="td_product['ribbon']['text_color']"/>
            <t t-set="text_color" t-value="text_color and ('color: %s;' % text_color) or ''"/>
```
so that we do not generate `color: ;` ?